File: leetcode/389.找不同/main.py
from collections import Counter
from functools import reduce
from operator import xor


class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        return chr(reduce(xor, map(ord, s + t)))

    def findTheDifference1(self, s: str, t: str) -> str:
        return list(Counter(t) - Counter(s))[0]

    def findTheDifference2(self, s: str, t: str) -> str:
        res = ord(t[-1])
        for i in range(len(s)):
            res += ord(t[i]) - ord(s[i])
        return chr(res)

    def findTheDifference3(self, s: str, t: str) -> str:
        diff = 0
        for c in t:
            diff += ord(c)
        for c in s:
            diff -= ord(c)
        return chr(diff)

File: leetcode/389.找不同/main.go
package main

func findTheDifference(s string, t string) byte {
	cnt := [26]int{}
	for _, ch := range s {
		cnt[ch-'a']++
	}
	for i := 0; ; i++ {
		ch := t[i]
		cnt[ch-'a']--
		if cnt[ch-'a'] < 0 {
			return ch
		}
	}
}

func findTheDifference1(s string, t string) byte {
	diff := 0

	for c := range s {
		diff -= int(s[c])
	}

	for c := range t {
		diff += int(t[c])
	}

	return byte(rune(diff))
}
